chapters/custom-license-namespaces.md

```diff
@@ -0,0 +1,194 @@
+# Annex Z Custom License Namespaces
+
+
+## Z.1 Rationale
+
+The
+[SPDX License List](https://spdx.org/licenses)
+is a curated list of commonly found licenses and exceptions
+used in free and open or collaborative software,
+data, hardware, or documentation.
+Licenses in the SPDX License List have stable license identifiers
+and canonical permanent URLs
+pointing to specified license texts,
+with markup and matching guidelines
+to capture non-substantive variations to the text.
+
+The SPDX specification also defines a mechanism
+for creators of SPDX documents
+to define custom license texts and identifiers.
+This enables SPDX documents to refer to licenses
+that might never be included on the SPDX License List,
+for a variety of reasons.
+These custom license identifiers are prefixed with `LicenseRef-`
+as defined in
+[Annex D - SPDX license expressions](./SPDX-license-expressions.md).
+This identifier format
+was originally assumed to be used
+in the context of an SPDX document,
+so that the license text for that
+identifier would be specified
+in a corresponding
+[Other licensing information detected](./other-licensing-information-detected.md) section.
+
+This annex defines mechanisms to use stable identifiers for custom licenses
+outside the context of a single SPDX document.
+For example, developers might wish to insert
+SPDX short-form license identifiers
+into their source code
+for licenses that are not on the SPDX License List,
+as described in
+[Annex E - Using SPDX license list short identifiers in source files](./using-SPDX-short-identifiers-in-source-files.md)
+Additionally,
+tools may want to interchange information about such licenses
+without having to use the same internal format
+to generate identifiers,
+or without having to compare license texts included in SPDX documents
+to confirm that they are in fact the same text.
+
+
+## Z.2 Brief overview
+
+This annex defines two formats and mechanisms
+for using the existing `LicenseRef-` format
+in a manner that enables stable identifier usage.
+Under this system:
+
+1. each license namespace consists of
+   either a standard order DNS name (such as `example.com`)
+   or an organization name (such as `ExampleCorp`)
+   that is under the control of the namespace maintainer's organization;
+2. the namespace maintainer creates
+   an SPDX document defining license texts
+   and corresponding identifiers
+   which include the namespave,
+   as described below;
+3. the namespace and the SPDX document
```

Comment:
See my main comments about whether the SPDX document should itself be registered with the SPDX project (in the sense of having the document itself checked into a repo within the SPDX GitHub org), vs. having them register a _creator-controlled URL_ with the SPDX project.